### ISSUE TITLE
Load dark theme at start before keyboard is visible

### ISSUE THREAD

**Issue by mc776** [dessalines/thumb-key#1020]
### Describe your suggested feature

1. Set up Thumb-Key to use a dark theme.
2. Kill the app or install an update.
3. Open the keyboard again.

Expected:
4. The keyboard opens with your theme.

Actual:
4. The keyboard dazzles you with the default light theme for about a second or so before loading your dark theme.

Is there a way to do it so the theme loads first before the keyboard is visible, or maybe add an optional (dark) splash screen?

### Other details

This happens whether I choose "System" or "Dark" as the theme.

I mostly use Matrix_AMOLED but it seems to be the same on Blue as well.

### Acknowledgements

- [X] I have written a short but informative title.
- [X] I have updated the app to **[the latest version](https://github.com/dessalines/thumb-key/releases/latest)**.
- [X] I have checked through the app settings for my feature.
- [X] I have searched the existing issues and this is a new one, **NOT** a  duplicate or related to another open issue.
- [X] This is a **single** feature request, in case of multiple feature  requests I will open a separate issue for each one (they can always link to each other if related)
- [X] This is not a question or a discussion, in which case I should have  gone to [lemmy.ml/c/thumbkey](https://lemmy.ml/c/thumbkey)
- [ ] I have admitted that I am a clown by having checked this box, as I have not read these acknowledgements. 🤡
- [X] I will fill out all of the requested information in this form.

**Comment by MarvinBaral** [comment]
i confirm the behaviour, also Matrix_AMOLED

**Comment by MarvinBaral** [comment]
For me its roughly 500ms (Android 8, Motorola Z Play)

You could change the hardcoded default theme to Matrix_AMOLED. I would also guess that it is way more popular than the white-red default. Do you know which theme Is most popular?

**Comment by dessalines** [comment]
The default theme is system dynamic, and that's most likely the most popular. The matrix one isn't even light-dark adapted.

**Comment by mc776** [comment]
> I tested and can confirm, but it's _really difficult_ to notice on my device, probably sub 50ms.
> 
> It needs to read from the DB to load your settings. I'd rather not block and slow down the initial loading of the keyboard.

I tried killing and restarting the keyboard just now to see if I could count how long exactly it was, but now it's only showing the default theme just long enough for me to notice it's different but not long enough for me to tell you what colour the text was if i didn't already know (so maybe ~100ms???) - the longer delay earlier might have been from something else that's since been fixed.

**Comment by mc776** [comment]
Back to 1 second as of 3.4.5.

**Comment by MarvinBaral** [comment]
> The default theme is system dynamic, and that's most likely the most popular. The matrix one isn't even light-dark adapted.

I had the same unwanted behaviour with the system dynamic keyboard theme configured on fored Dark Mode (initially Light with default keys, and then Dark with configured keys).

A issue source might be that in my Android 8 I can't set a system wide theme so it is fixed on Light Mode. I manually have to set every app to Dark Mode, (which you thankfully can do here).

So maybe in the initial startup state thumb-key has the dark-light adjustment based on the system (always Light Mode for me) and only after the DB read recognizes the forced Dark Mode app setting.

I would suggest to make the initial state recognize the app setting for light-dark, but that will probably require a DB read which is not the initial state anymore. And this issue doesn't come up that often and I'm by now already used to it. So I myself would be ok with a Wontfix on that issue.
